File: src/main/java/pl/tsimur/jpatest/exception/NotEnoughMoneyException.java
package pl.tsimur.jpatest.exception;

import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.server.ResponseStatusException;

import static org.springframework.http.HttpStatus.PAYMENT_REQUIRED;

@ResponseStatus(PAYMENT_REQUIRED)
public class NotEnoughMoneyException extends ResponseStatusException {
    public NotEnoughMoneyException(Number balance) {
        super(PAYMENT_REQUIRED, "💸 Not enough money. Current balance: " + balance);
    }
}

File: src/main/kotlin/pl/timur/jpatest/exception/NotEnoughMoneyException.kt
package pl.timur.jpatest.exception

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.server.ResponseStatusException

@ResponseStatus(HttpStatus.PAYMENT_REQUIRED)
class NotEnoughMoneyException(balance: Number) :
    ResponseStatusException(HttpStatus.PAYMENT_REQUIRED, "💸 Not enough money. Current balance: $balance")